### PR TITLE
Android gecko profiling

### LINKED ISSUE #13: Teach browsertime to collect Gecko Profiler output on Android
#3 added support for the Gecko Profiler to browsertime, but it wasn't until #2 that Android support landed.  This ticket tracks teaching browsertime to collect the Gecko Profiler outputs and get them into the reports.

### PR DESCRIPTION
**Pull request by dpalmeiro**
Add Android support for gecko profiling and update the sampling interval based on platform.  Fixes https://github.com/mozilla/browsertime/issues/13.

### CODE REVIEW

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 1]
```diff
@@ -37,6 +38,17 @@ class FirefoxDelegate {
       let threadString = '["' + chosenThreads.join('","') + '"]';
 
       let interval = this.firefoxConfig.geckoProfilerParams.interval;
+
+      // Set platform specific sampling intervals if not explicitly specified.
+      if (
+        this.firefoxConfig.geckoProfiler &&
+        !this.firefoxConfig.geckoProfilerParams.interval
+      ) {
+        interval = this.firefoxConfig.android
```
Comment (ncalexan) [inline lib/firefox/webdriver/firefoxDelegate.js]: I'd gently prefer `isAndroidConfigured`, since this is what it's for.

File: lib/support/cli.js [hunk 2]
```diff
@@ -283,7 +283,6 @@ module.exports.parseCommandLine = function parseCommandLine() {
     })
     .option('firefox.geckoProfilerParams.interval', {
       describe: 'Sampling interval in ms.',
```
Comment (ncalexan) [inline lib/support/cli.js]: Consider including conditional default in the description, although that can itself get out of date.

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 3]
```diff
@@ -75,24 +78,38 @@ class FirefoxDelegate {
...
+      let profileDir = await this.storageManager.createSubDataDir(
+        pathToFolder(result.url, this.options)
+      );
+      let destinationFilename = path.join(
+        profileDir,
         `geckoProfile-${index}.json`
       );
 
+      let deviceProfileFilename = destinationFilename;
+      if (this.options.firefox.android) {
```
Comment (ncalexan) [inline lib/firefox/webdriver/firefoxDelegate.js]: Again, `isAndroidConfigured`.

File: lib/firefox/webdriver/firefoxDelegate.js [hunk 4]
```diff
@@ -75,24 +78,38 @@ class FirefoxDelegate {
...
+
       // Must use String.raw or else the backslashes on Windows will be escapes.
       const script = `
       var callback = arguments[arguments.length - 1];
-       Services.profiler.dumpProfileToFileAsync(String.raw\`${profileFilename}\`)
+       Services.profiler.dumpProfileToFileAsync(String.raw\`${deviceProfileFilename}\`)
         .then(callback)
         .catch((e) => callback({'error' : e}));
        `;
+      log.info('Stop GeckoProfiler.');
```
Comment (ncalexan) [inline lib/firefox/webdriver/firefoxDelegate.js]: This should go after the `Collect GeckoProfiler` invocation.

**Review by ncalexan — APPROVED** [review]
Just nits, no need for re-review.  Thanks for doing this!

File: lib/support/cli.js (hunk 2 above)
Comment (dpalmeiro) [reply lib/support/cli.js]: I'm not sure if we can be certain the target platform is android at this point yet.

Comment (ncalexan) [reply lib/support/cli.js]: > I'm not sure if we can be certain the target platform is android at this point yet.

Right -- I mean, `describe: 'Sampling interval in ms; defaults to 1ms on Desktop and 4ms on Android.'`.

**Pull request merged**